### PR TITLE
[第4組]Refactor naming convetions to improve code readability

### PR DESCRIPTION
**Pull request by yozen0405**
### Purpose

This pull request aims to improve the overall readability of the project by enhancing variable and function names, modularizing some functions, and ensuring that the code follows clean code principles.

### Changes

<details>
  <summary>local variables</summary>
  
- l, a → removed: The intermediate variables l and a were removed, and the data is now directly appended to all_news_data using extend, simplifying the logic.
- p → page: Renamed in pagination logic to clarify that it refers to the page number.
- p2 → request_params: Renamed p2 to request_params for clearer understanding of the variable’s purpose, improving code readability.
- ai → ai_response: Makes it clear that the variable holds the response from the AI model.
- title → article_title or detailed_title: Clarifying the variable’s content.
- result → summary_result: Describes that the result is a summary.
- n → username: Updated to a more meaningful variable name to clarify that the function parameter is the user’s username, adhering to clean code principles.
- OuO → user: In the check_user_password_is_correct function, to clarify that this refers to a User object.
- n → article: Renamed in local variable to indicates the item in each iteration.
- cnt → upvoted_count, id → article_id: Standardized for clarity in upvote and news article-related functions.
- m → relevance_check_prompt, summary_prompt, summary_generation_prompt: Consolidated renaming of the prompt variable used for chat interactions with GPT, based on its role in relevance checking and summary generation.
- p2 → request_params: Changed in the function where API request parameters are built for fetching news data.

</details>

<details>
  <summary>global variables</summary>

- Replaced hard-coded magic numbers like 50, 200 with clearly defined constants (MaxUsernameSize, MaxPasswordSize) for better maintainability.
- bgs → scheduler: Renamed the background scheduler for better readability and clarity of its purpose.

</details>

<details>
  <summary>params</summary>

- u → user: Used in the functions where user information is being passed, such as in read_user_news, upvote_article, news_summary.
- id → article_id: Updated to indicate that the variable holds the ID of a news article, used in query parameters for links.
- id2 → article_id: Replaced with article_id to maintain consistency and clarity in identifying that the variable represents the article’s ID in functions like news_exists.
- p1 → plain_password, p2 → hashed_password: Clarified the parameter in the password verification function to specify that it refers to the plain-text password and hashed_password.
- n_id, u_id → article_id, user_id: Standardized the use of article_id, user_id for consistency in function like toggle_upvote.

</details>

<details>
  <summary>functions</summary>

- get_new_info → fetch_news_articles_by_keyword: Changed to clarify that the function fetches news articles based on a search term.
- get_new → fetch_and_process_news: Modified to better describe that the function fetches and processes the news articles.
- add_new → add_news_article: Clarified the purpose of adding a new news article to the database.
- toggle_upvote → toggle_article_upvote: Updated to make it clear that the function specifically handles the upvote status for a news article.
- read_news, read_user_news → fetch_news_with_upvote_details, get_user_specific_news: These two functions were renamed because their purposes were too similar, violating clean code principles. By renaming them, it clarifies that one function fetches general news with upvote details, while the other focuses on user-specific news, making their roles distinct.
- verify → verify_password: Renamed the function for clarity, making it more descriptive, ensures the function name clearly conveys its purpose.
- check_user_password_is_correct → validate_user_credentials: Renamed the function to better reflect its purpose of both querying the user from the database and validating the user’s credentials (username and password), which aligns more closely with clean code principles.

</details>

### CODE REVIEW

**Comment by siangting** [comment]
The comment on the PR is excellent, well-organized, and thoroughly categorized.

**Comment by Steven0811** [comment]
Your variables' names are refactored in a good way, and makes me understand the meaning of them easily. The comment on the PR is also excellent. But some "db" have not been renamed.

**Comment by Rogue8464** [comment]
function跟variable都名稱都命名的很清楚，有明確的表達意義且不會過於冗長，也有把magic number提取出來，讓程式整體的可讀性提高了很多

**Comment by percykugood** [comment]
命名都改的讓人可以很好的理解，而且也都寫得很清楚，這讓我很好的閱讀並理解這段程式

**Comment by Yi-Cheng-Wang** [comment]
**Evaluation:**
- High degree of variable refactoring
- Variable renaming is good
- Code layout is particularly tidy
- Commits are good
- Clarity of the PR is high

**Additional Evaluation:**
- The handling of magic numbers on lines 37–38 is very good; it is one of the few instances where this has been addressed!

**Suggestions:**
- There are still magic numbers that can be addressed; for example, the value `1892dhianiandowqd0n` on line 290 could be passed in a better way.
- Line 52 has excessive whitespace, which causes Git to consider it a change.
- The variable `expires_delta` on line 295 is not easily understandable and should be changed.

**Comment by mlcp29th-hi** [comment]
Hello,
1. 依照 [PEP 8 Style Guideline](https://peps.python.org/pep-0008/)，想建議 `MaxUsernameSize`、`MaxPasswordSize` 等固定不變的常數應以 `UPPER_CASE` 的形式命名，如 `MAX_USERNAME_SIZE`、`MAX_PASSWORD_SIZE`，另外值得討論的是此處是否有提取 magic number 的必要：
Screenshot

提取 magic number 能提供更詳細的上下文以幫助其他工程師了解該數值的意義，然而 `String` 初始化函式已經表明該參數代表欄位的長度上限，結合該屬性名稱 `username`，我們能一眼看出這是一個代表「使用者名稱」、「長度上限為 50 字元的字串欄位」，再加上目前僅有一處使用此數值，或許可以不必修改。

2. Line 52 行尾意外新增多餘的空白，建議移除。
3. `schedulers` 存放的是單個 `BackgroundScheduler` 物件，建議以單數命名即可，可修改為 `scheduler`。
4. 建議命名時若有更好的選擇，最好盡量避免 data/info/result 等形式，否則以旁人的角度較不容易立刻理解 `news_article`、`news_article_data`、`all_news_data`、`result`等的意義／差別，畢竟變數本身即資料，`_data` 等後綴對於理解其意義的幫助有限。
5. 部分程式在處理新聞資料時會用 news/article/news_article 命名，建議統一個名稱以維持一致性。
6. `delete_stmt`、`insert_stmt` 可以考慮改成 `delete_statement`、`insert_statement` 使人們討論此程式時便於發音，較利於溝通。

File: backend/main.py [hunk 1]
```diff
@@ -34,22 +34,22 @@
 
 # from pydantic import BaseModel
 
-
+MaxUsernameSize = 50 
+MaxPasswordSize = 200
```
Comment (taro0513) [inline backend/main.py:38]: Constants are usually written in all capital letters with underscores separating words.

File: backend/main.py [hunk 2]
```diff
@@ -120,168 +116,158 @@ class NewsArticle(Base):
...
-        relevance = ai.choices[0].message.content
+        relevance = ai_response.choices[0].message.content
         if relevance == "high":
-            response = requests.get(news["titleLink"])
+            response = requests.get(article["titleLink"])
             soup = BeautifulSoup(response.text, "html.parser")
             # 標題
-            title = soup.find("h1", class_="article-content__title").text
-            time = soup.find("time", class_="article-content__time").text
+            detailed_title = soup.find("h1", class_="article-content__title").text
```
Comment (taro0513) [inline backend/main.py:205]: What's the difference between "title" and "detailed_title" what is the information that you want to give?

**Review by taro0513 — COMMENTED** [review]
LGTM

**Pull request merged**